### DISCUSSION TITLE
How can i debug app in lowdefy?

### DISCUSSION THREAD

**Discussion by awsomedesigner** [lowdefy/lowdefy#1182] [Questions and Help] [answered]
I want to debug the app. what's going wrong with my code in lowdefy app. 
How can I debug any error?

**Comment by 13316090972** [comment, marked as answer]
I know 3 ways to debug your app:
1. use[ _log](https://docs.lowdefy.com/_log) operartor to print your [state](https://docs.lowdefy.com/context-and-state) on browser console.
2. if you run your app on localhost, you can input lowdefy.contexts on your browser console to see all of your states.
3. read the built json file in the .lowdefy/build directory under your app directory.It will be helpful in template files.
